.github/workflows/tests.yml

```diff
@@ -31,29 +31,43 @@ jobs:
           - ubuntu-latest
           - macos-latest
           - windows-latest
-        python-version:
-          - "3.7"
-          - "3.8"
-          - "3.9"
-          - "3.10"
-          - "3.11"
-          - "pypy-3.7"
```

Comment:
This section of the matrix is removed because the `setup-python` action can now install _all_ of these versions simultaneously, and Tox can take detect and run all of these in one go.

This reduces the matrix down from a cross-product of operating systems and Python versions (18x runs) to merely a list of three operating systems (3x runs).

The advantage here is that slow, duplicate operations that must happen just to get the operating system environment to a functional state now happen once per operating system (3x) instead of once per OS _per Python version_ (18x).

For example, looking at [a recent run of the tests against the `main` branch](https://github.com/nedbat/scriv/actions/runs/4086514545/jobs/7045891295), these are steps that had to happen multiple times just to get to a usable state where Tox could run:

* Checkout the repo (13s)
* Set up Python (15s)
* Install dependencies (18s)
* Install pandoc (31s)

Moving these six Python versions from the matrix to the `setup-python` action eliminates this time multiplier.